pkg/registry/registry.go

```diff
@@ -26,30 +25,43 @@ type Registry interface {
 	StopRegistry() error
 }
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
```

Comment:
This looks very much like [http.RoundTripper](https://pkg.go.dev/net/http#RoundTripper).

Reply:
How is it better?